homeassistant/components/onewire/config_flow.py

```diff
@@ -0,0 +1,112 @@
+"""Config flow for 1-Wire component."""
+import logging
+import os
+
+from pyownet import protocol
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TYPE
+
+from .const import (  # pylint: disable=unused-import
+    CONF_MOUNT_DIR,
+    CONF_TYPE_OWFS,
+    CONF_TYPE_OWSERVER,
+    CONF_TYPE_SYSBUS,
+    DEFAULT_OWFS_MOUNT_DIR,
+    DEFAULT_OWSERVER_HOST,
+    DEFAULT_OWSERVER_PORT,
+    DEFAULT_SYSBUS_MOUNT_DIR,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OneWireFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle 1-Wire config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize 1-Wire config flow."""
+        self.onewire_config = {}
+
+    async def async_step_user(self, user_input=None):
```

Comment:
how are we handling duplicate config entry detection? any unique ids? may need to do checks against existing entries for similiar configs.

Reply:
Add in 9f55b69255e3ac78b38489c5c9a6f4ee9f927546